http-agent/src/main/java/org/motechproject/http/agent/listener/HttpClientEventListener.java

```diff
@@ -186,7 +186,21 @@ public void handle(MotechEvent motechEvent) {
 
                 LOGGER.info("Posting Http request -- Url: {}, Data: {}",
                         url, String.valueOf(requestData));
-                ResponseEntity<?> response = doExecuteForReturnType(url, requestData, method, restTemplate);
+                ResponseEntity<?> response = null;
+                String errorMessage = null;
+                try {
+                    response = doExecuteForReturnType(url, requestData, method, restTemplate);
+                } catch (Exception e) {
```

Comment:
I tried but got error: exception "HttpException is never thrown in body of corresponding try statment" while compilation